ui/analytics.js

```diff
@@ -76,7 +76,149 @@ if (window.localStorage.getItem(SHARE_INTERNAL) === 'true') internalAnalyticsEna
 // if (window.localStorage.getItem(SHARE_THIRD_PARTY) === 'true') thirdPartyAnalyticsEnabled = true;
 // @endif
 
+/**
+ * Determine the mobile device type viewing the data
+ * This function returns one of 'and' (Android), 'ios', or 'web'.
+ *
+ * @returns {String}
+ */
+function getDeviceType() {
+  var userAgent = navigator.userAgent || navigator.vendor || window.opera;
+
+  if (/android/i.test(userAgent)) {
+    return 'and';
+  }
+
+  // iOS detection from: http://stackoverflow.com/a/9039885/177710
+  if (/iPad|iPhone|iPod/.test(userAgent) && !window.MSStream) {
+    return 'ios';
+  }
+
+  // default as web, this can be optimized
+  return 'web';
+}
+
+let durationInSeconds = 10;
+let amountOfBufferEvents = 0;
+let amountOfBufferTimeInMS = 0;
+let videoType, userId, claimUrl, playerPoweredBy, timeAtBuffer, videoPlayer;
+
+async function sendAndResetWatchmanData() {
+  var protocol;
+  if (videoType === 'application/x-mpegURL') {
+    protocol = 'hls';
+  } else {
+    protocol = 'stb';
+  }
+
+  console.log(claimUrl);
+
+  timeAtBuffer = Math.round(videoPlayer.currentTime()) * 1000;
+
+  var totalDurationInSeconds = Math.round(videoPlayer.duration());
+
+  const objectToSend = {
+    rebuf_count: amountOfBufferEvents,
+    rebuf_duration: amountOfBufferTimeInMS,
+    url: claimUrl.replace('lbry://', ''),
+    device: getDeviceType(),
+    duration: Math.round(durationInSeconds) * 1000,
+    protocol,
+    player: playerPoweredBy,
+    user_id: userId.toString(),
+    position: Math.round(timeAtBuffer),
+    rel_position: Math.round((timeAtBuffer / (totalDurationInSeconds * 1000)) * 100),
+  };
+
+  await sendWatchmanData(objectToSend);
+
+  amountOfBufferEvents = 0;
+  amountOfBufferTimeInMS = 0;
+  timeAtBuffer = null;
+}
+
+let watchmanInterval;
+function stopWatchmanInterval() {
+  console.log('turning off watchman interval');
+  clearInterval(watchmanInterval);
+  watchmanInterval = null;
+}
+function startWatchmanIntervalIfNotRunning() {
+  console.log('turning on watchman interval');
+  if (!watchmanInterval) {
+    console.log('watchman interval turned back on');
+    watchmanInterval = setInterval(sendAndResetWatchmanData, 1000 * durationInSeconds);
+  }
+}
+
+async function sendWatchmanData(body) {
+  try {
+    const response = await fetch('https://watchman.na-backend.odysee.com/reports/playback', {
```

Comment:
I think we can put this URL somewhere - maybe config, maybe a constant. Preferably a config.